File: 1894-merge-strings-alternately/merge-strings-alternately.cpp
class Solution {
public:
    string mergeAlternately(string word1, string word2) {
        int totalLength=word1.size()+word2.size();
        string newS(totalLength,' ');
        int i=0;
        int j=0;
        int k=0;
        while(i<totalLength){
            if(j<word1.size()){
                newS[i]=word1[j];
                j++;
                i++;
            }
            if(k<word2.size()){
                newS[i]=word2[k];
                k++;
                i++;
            }
        }
        return newS;
    }
};

File: 1894-merge-strings-alternately/merge-strings-alternately.java
class Solution {
    public String mergeAlternately(String word1, String word2) {
        String ans="";
        int minLen=Math.min(word1.length(),word2.length());
        int i=0;
        while(i<minLen){
            if(i>=word1.length()) break;
            else ans+=word1.charAt(i);
            if(i>=word2.length()) {
            break;
            }
            else ans+=word2.charAt(i);
            i++;
        }
        if(i<word1.length()) ans+=word1.substring(i,word1.length());
        if(i<word2.length()) ans+=word2.substring(i,word2.length());
        return ans;
    }
}